packages/server/src/index.ts

```diff
@@ -1036,6 +1039,55 @@ export class App {
             await this.buildChatflow(req, res, undefined, true, true)
         })
 
+        // ----------------------------------------
+        // Prompt from Hub
+        // ----------------------------------------
+        this.app.post('/api/v1/load-prompt', async (req: Request, res: Response) => {
+            try {
+                const credential = await this.AppDataSource.getRepository(Credential).findOneBy({
+                    id: req.body.credential
+                })
+
+                if (!credential) return res.status(404).json({ error: `Credential ${req.body.credential} not found` })
+
+                // Decrypt credentialData
+                const decryptedCredentialData = await decryptCredentialData(credential.encryptedData, credential.credentialName, undefined)
+                let hub = new Client({ apiKey: decryptedCredentialData.langsmithApiKey, apiUrl: decryptedCredentialData.langsmithEndpoint })
+                const prompt = await hub.pull(req.body.promptName)
+                const templates = parsePrompt(prompt)
+
+                return res.json({ status: 'OK', prompt: req.body.promptName, templates: templates })
+            } catch (e: any) {
+                return res.json({ status: 'ERROR', prompt: req.body.promptName, error: e?.message })
+            }
+        })
+
+        this.app.post('/api/v1/prompts-list', async (req: Request, res: Response) => {
+            try {
+                const credential = await this.AppDataSource.getRepository(Credential).findOneBy({
+                    id: req.body.credential
+                })
+
+                if (!credential) return res.status(404).json({ error: `Credential ${req.body.credential} not found` })
+                // Decrypt credentialData
+                const decryptedCredentialData = await decryptCredentialData(credential.encryptedData, credential.credentialName, undefined)
+
+                const headers = {}
+                // @ts-ignore
+                headers['x-api-key'] = decryptedCredentialData.langsmithApiKey
+
+                const tags = req.body.tags ? `tags=${req.body.tags}` : ''
+                const url = `https://web.hub.langchain.com/repos/?${tags}offset=0&limit=20&has_commits=true&sort_field=num_likes&sort_direction=desc&is_archived=false`
```

Comment:
do we need offset and limit here? on the UI we dont have pagination right?

Reply:
Not needed for now, fixed and pushed changes